Bigtable/src/BigtableClient.php

```diff
@@ -0,0 +1,205 @@
+     *     @type callable $authHttpHandler A handler used to deliver Psr7
+     *           requests specifically for authentication.
+     *     @type FetchAuthTokenInterface $credentialsFetcher A credentials
+     *           fetcher instance.
+     *     @type array $keyFile The contents of the service account credentials
+     *           .json file retrieved from the Google Developer's Console.
+     *           Ex: `json_decode(file_get_contents($path), true)`.
+     *     @type string $keyFilePath The full path to your service account
+     *           credentials .json file retrieved from the Google Developers
+     *           Console.
+     *     @type float $requestTimeout Seconds to wait before timing out the
+     *           request. **Defaults to** `0` with REST and `60` with gRPC.
+     *     @type int $retries Number of retries for a failed request.
+     *           **Defaults to** `3`.
+     * }
+     *
+     * @throws GoogleException If the gRPC extension is not enabled.
+     */
+    public function __construct(array $config = [])
+    {
+        $this->requireGrpc();
+        $config += [
+            'scopes' => [
+                self::FULL_CONTROL_SCOPE,
+                self::ADMIN_SCOPE
+            ],
+            'projectIdRequired' => true
+        ];
+        $this->connection = new Grpc($this->configureAuthentication($config));
+        $this->setLroProperties(new LongRunningConnection($this->connection), [
+            [
+                'typeUrl' => 'type.googleapis.com/google.bigtable.admin.instance.v2.UpdateInstanceMetadata',
+                'callable' => function ($instance) {
+                    $name = InstanceAdminClient::parseName($instance['name'])['instance'];
+                    return $this->instance($name, $instance);
+                }
+            ], [
+                'typeUrl' => 'type.googleapis.com/google.bigtable.admin.instance.v2.CreateInstanceMetadata',
+                'callable' => function ($instance) {
+                    $name = InstanceAdminClient::parseName($instance['name'])['instance'];
+                    return $this->instance($name, $instance);
+                }
+            ]
+        ]);
+    }
+
+    /**
+     * Lazily instantiate an instance.
+     *
+     * Example:
+     * ```
+     * $instance = $bigtable->instance('my-instance');
+     * ```
+     *
+     * @param string $instanceId The instance ID
+     * @param array $instance [optional] {
+     *     Configuration options
+     *
+     *     @type string $displayName **Defaults to** the value of $instanceId.
+     *     @type int $instanceType Possible values are represented by the following constants:
+     *           `Google\Cloud\Bigtable\Instance::INSTANCE_TYPE_PRODUCTION` and
+     *           `Google\Cloud\Bigtable\Instance::INSTANCE_TYPE_DEVELOPMENT`.
+     *           **Defaults to** using `Google\Cloud\Bigtable\Instance::INSTANCE_TYPE_UNSPECIFIED`.
```

Comment:
@jdpedrie UNSPECIFIED is a valid protobuf value. By providing UNSPECIFIED we are letting server decide what should be the default value. Currently its PRODUCTION, in future it can be changed to DEVELOPMENT, hence its better to keep UNSPECIFIED rather than PRODUCTION.

Reply:
A few thoughts on this:

- Changing a default value from `PRODUCTION` to `DEVELOPMENT` strikes me as a breaking change, as existing user code would alter in behavior. I'm certainly open to thoughts on that perspective. Assuming that understanding is correct - is that a scenario we really need to be concerned with? A major version bump would almost certainly come with other updates to the client we need to parse through anyways.
- Is there any difference between `UNSPECIFIED` and just omitting the optional argument? Based on my testing there isn't, so I'm not sure why we would inform our users that whatever instance is created will be `UNSPECIFIED`. Wouldn't it be helpful to share with them what it *actually* will be once created?

Reply:
I am ok with omitting optional parameter altogether. Protobuf will set to UNSPECIFIED value anyway.